src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java

```diff
@@ -358,28 +367,63 @@ public void validate() {
 
     /**
      * Does this feed version have any critical errors that would prevent it being loaded to OTP?
-     * @return whether feed version has critical errors
+     * @return whether the feed version has any critical errors
      */
     public boolean hasCriticalErrors() {
-        return hasCriticalErrorsExceptingDate() ||
-                validationResult.lastCalendarDate == null ||
-                (LocalDate.now()).isAfter(validationResult.lastCalendarDate);
+        return hasValidationAndLoadErrors() ||
+            hasFeedVersionExpired() ||
+            hasHighSeverityErrorTypes();
     }
 
     /**
-     * Does this feed have any critical errors other than possibly being expired?
-     * @return whether feed version has critical errors (outside of expiration)
+     * Does this feed have any validation or load errors?
+     * @return whether feed version has any validation or load errors
      */
-    private boolean hasCriticalErrorsExceptingDate() {
+    private boolean hasValidationAndLoadErrors() {
         if (validationResult == null)
             return true;
-
         return validationResult.fatalException != null ||
             !validationSummary().bounds.areValid() ||
             feedLoadResult.stopTimes.rowCount == 0 ||
             feedLoadResult.trips.rowCount == 0 ||
             feedLoadResult.agency.rowCount == 0;
+    }
+
+    /**
+     * Has this feed expired?
+     * @return whether feed version is outside of expiration
+     */
+    private boolean hasFeedVersionExpired() {
+        return validationResult.lastCalendarDate == null ||
+            LocalDate.now().isAfter(validationResult.lastCalendarDate);
+    }
 
+    /**
+     * Has this feed version produced any high severity error types when being validated?
+     * @return whether high severity error types have been flagged.
+     */
+    private boolean hasHighSeverityErrorTypes() {
+        Set<String> highSeverityErrorTypes = Stream.of(NewGTFSErrorType.values())
+                .filter(type -> type.priority == Priority.HIGH)
```

Comment:
I've seen a bunch of transit agencies with very rural service have HIGH priority errors for valid stops that aren't near population centers. This will affect the auto-deployment. Perhaps there is a way to make an exception for some of the high-severity errors that shouldn't result in OTP encountering an error. I'm thinking the following might be OK to ignore: `ROUTE_UNUSED`, `STOP_GEOGRAPHIC_OUTLIER` and `STOP_LOW_POPULATION_DENSITY`.

Reply:
Updated.